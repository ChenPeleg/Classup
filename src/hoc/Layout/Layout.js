import React, { Component } from "react";
import axios from "axios";
import LearningContainer from "../../containers/LearningContainer/LearningContainer";
import Navigation from "../../components/Navigation/Navigation";
import "./Layout.scss";
import LoaderAnimation from "../../components/LoaderAnimation/LoaderAnimation";

class Layout extends Component {
  state = { AllQuestions: false, soundOn: false };
  soundHandler = () => {
    this.setState({ ...this.state, soundOn: !this.state.soundOn });
  };
  componentDidMount() {
    let url =
      "https://click123.s3.eu-west-2.amazonaws.com/classUp_dev/allquestions.json";
    url = "allquestions.json"; // for local file
    axios.get(url, { crossdomain: true }).then((res) => {
      this.setState({ ...this.state, AllQuestions: res.data });
    });
  }
  render() {
    return (
      <React.Fragment>
        <Navigation
          AllQuestions={
            this.state.AllQuestions
              ? this.state.AllQuestions
              : { meta: { name: "..." } }
          }
          soundOn={this.state.soundOn}
          soundHandler={this.soundHandler}
        />

        {this.state.AllQuestions ? (
          <LearningContainer
            AllQuestions={this.state.AllQuestions}
            soundOn={this.state.soundOn}
          />
        ) : (
            <LoaderAnimation />
          )}
      </React.Fragment>
    );
  }
}
export default Layout;
