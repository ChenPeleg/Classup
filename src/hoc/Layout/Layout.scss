// @import "../../assets/styles/base";

@font-face {
  font-family: "KulimPark";
  font-weight: 400;
  font-style: normal;
  src: url("../../assets/fonts/Kulim_Park/KulimPark-Regular.ttf")
    format("truetype");
}

.Layout {
  font-family: KulimPark;
  position: absolute;
  height: 100%;
  width: 80%;
  left: 10%;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  border-radius: 0.5%;
  text-align: center;
  font-size: 26px;
  background: rgb(2, 0, 36);
  background: linear-gradient(
    90deg,
    rgba(2, 0, 36, 1) 0%,
    rgba(12, 134, 196, 0.9868989832261029) 0%,
    rgba(0, 212, 255, 1) 100%
  );
}
@media only screen and (max-width: 600px) {
  .Layout {
    width: 100%;
    left: 0%;
  }
}

.Layout-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.Layout-title {
  font-size: 1.5em;
}

.Layout-intro {
  font-size: large;
}

@keyframes Layout-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
