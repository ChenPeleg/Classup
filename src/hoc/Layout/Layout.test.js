import React from "react";
import { shallow } from "enzyme";
import Layout from "./Layout";
import renderer from 'react-test-renderer';
const wrap = shallow(<Layout />)

describe("<Layout>", () => {
  it("renders correctly", () => { expect(wrap).toMatchSnapshot() })
  it("renders correctly", () => { expect(wrap).toMatchSnapshot() })
})