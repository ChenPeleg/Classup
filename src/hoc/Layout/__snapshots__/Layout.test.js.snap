// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> renders correctly 1`] = `
<Fragment>
  <Navigation
    AllQuestions={
      Object {
        "meta": Object {
          "name": "...",
        },
      }
    }
    soundHandler={[Function]}
    soundOn={false}
  />
  <LoaderAnimation />
</Fragment>
`;

exports[`<Layout> renders correctly 2`] = `
<Fragment>
  <Navigation
    AllQuestions={
      Object {
        "meta": Object {
          "name": "...",
        },
      }
    }
    soundHandler={[Function]}
    soundOn={false}
  />
  <LoaderAnimation />
</Fragment>
`;
